### PR TITLE
Create allow-INTERPETER.inc

### PR DESCRIPTION
**Pull request by rusty-snake**
* allow-lua.inc
 * allow-perl.inc
 * allow-python2.inc
 * allow-python3.inc
 * allow-java.inc

See https://github.com/netblue30/firejail/pull/2735#discussion_r289574610

### CODE REVIEW

**Comment by SkewedZeppelin** [comment]
I think I got most of them.
I left out the template to not break #2735

**Comment by rusty-snake** [comment]
HEADER
COMMENTS
BLACKLISTS
NOBLACKLISTS
**ALLOW INCLUDES**
DISABLE INCLUDES
MKDIRS
WHITELISTS
WHITELIST INCLUDES
OPTIONS (no*)
PRIVATE OPTIONS (disable-mnt, private-*)
SPECIAL OPTIONS (mdwx, noexec, read-only)
REDIRECT INCLUDES

**Comment by rusty-snake** [comment]
meld, natron, qutebrowser, ranger need to be updated.

**Comment by SkewedZeppelin** [comment]
That should be it.
Aside from placement.

File: etc/JDownloader.profile [hunk 1]
```diff
@@ -8,7 +8,8 @@ include globals.local
 
 noblacklist ${HOME}/.jd
 
-# Allow access to java
+# Allow java (blacklisted by disable-interpreters.inc)
```
Comment (rusty-snake) [inline etc/JDownloader.profile]: java is blacklisted in disable-devel.

**Comment by Vincent43** [comment]
```
HEADER
COMMENTS
BLACKLISTS
NOBLACKLISTS
ALLOW INCLUDES
```
I think NOBLACKLISTS + ALLOW INCLUDES should be before BLACKLISTS otherwise they won't work :smiley:

**Comment by rusty-snake** [comment]
@Vincent43 to have the following in a profile make no sense:
```
noblacklist ${HOME}/something
blacklist ${HOME}/something
```
BLACKLIST means `blacklist` commands in the profile.

**Comment by glitsj16** [comment]
To accomodate [Arch Linux's Perl policy](https://wiki.archlinux.org/index.php/Perl_Policy) I'm wondering whether we should add

```
noblacklist ${PATH}/site_perl
noblacklist ${PATH}/vendor_perl
```

to the new [allow-perl.inc](https://github.com/netblue30/firejail/blob/c62fcf782117718c1abc558e392a12a74c798335/etc/allow-perl.inc).

**Comment by Fred-Barclay** [comment]
@glitsj16 I think we should

**Comment by rusty-snake** [comment]
@glitsj16 feel free to commit.

**Comment by Vincent43** [comment]
> to have the following in a profile make no sense:
> 
> noblacklist ${HOME}/something
> blacklist ${HOME}/something
> 

The opposite order also doesn't make sense. It doesn't make sense having noblacklist and blacklist for the same path in profile in general but noblacklist rule should precede blacklist one to be effective elsewhere and I think keeping that order here wold be consistent.

**Comment by rusty-snake** [comment]
@Vincent43 to do that to have a consistent/consequent order sounds good. I use this order above because every profile with `blacklist` and `noblacklist` has this order.
`grep "^blacklist" --context=5 /etc/profile/*.profile`

OK then lets have the following:

HEADER
_COMMENTS -- don't know what this is._
**IGNORES**
NOBLACKLISTS
**ALLOW INCLUDES**
**BLACKLISTS**
DISABLE INCLUDES
MKDIRS
WHITELISTS
WHITELIST INCLUDES
OPTIONS (aa, caps, ipc-namespace, machine-id, net*, no*, protocol, seccomp, shell, tracelog)
PRIVATE OPTIONS (disable-mnt, private*)
SPECIAL OPTIONS (env, mdwx, noexec, read-only, join-or-start)
REDIRECT INCLUDES

@glitsj16 where do we `blacklist ${PATH}/site_perl` & `blacklist ${PATH}/vendor_perl` ?

**Comment by glitsj16** [comment]
@rusty-snake At present we don't blacklist site_perl and vendor_perl, although I think we should. I forgot that part, nice catch. To keep this draft PR clean, would you be so kind as to add those to the current /etc/firejail/disable-interpreters.inc?

**Comment by SkewedZeppelin** [comment]
@rusty-snake 
> COMMENTS -- don't know what this is.

some profiles have comments on their use
eg.
https://github.com/netblue30/firejail/blob/master/etc/Xephyr.profile
https://github.com/netblue30/firejail/blob/master/etc/xpra.profile
https://github.com/netblue30/firejail/blob/master/etc/spectre-meltdown-checker.profile

**Comment by SkewedZeppelin** [comment]
Ignoring placement, is this fine to merge?

**Comment by rusty-snake** [comment]
@SkewedZeppelin I think so.

**Pull request merged**